src/ProjectFileManager.cpp

```diff
@@ -1397,16 +1421,24 @@ void ReactOnMusicFileImport(
          {
             AdornedRulerPanel::Get(*pProj).SetTimeDisplayMode(
                TimeDisplayMode::BeatsAndMeasures);
-            ProjectTimeSignature::Get(*pProj).SetTempo(syncInfo.rawAudioTempo);
+            auto& projTimeSignature = ProjectTimeSignature::Get(*pProj);
+            projTimeSignature.SetTempo(syncInfo.rawAudioTempo);
+            if (syncInfo.timeSignature.has_value())
+            {
+               projTimeSignature.SetUpperTimeSignature(
+                  MIR::GetNumerator(*syncInfo.timeSignature));
+               projTimeSignature.SetLowerTimeSignature(
+                  MIR::GetDenominator(*syncInfo.timeSignature));
+            }
          }
          else if (isBeatsAndMeasures)
             clip->StretchBy(syncInfo.stretchMinimizingPowOfTwo);
          if (!ans.readFromPreference && ans.yes)
             UndoManager::Get(*pProj).PushState(
```

Comment:
This may be a problem:  The function may take paths that do change the tracks, but do not push an undo state.  Therefore those changes of the tracks will be lost if you Undo the Import command that was previously pushed, then redo again.  Or the changes will be consolidated as part of the next Undo item that is pushed, which may be for an unrelated editing operation.